core/src/test/java/com/linecorp/armeria/server/streaming/JsonLinesTest.java

```diff
@@ -0,0 +1,160 @@
+import com.fasterxml.jackson.databind.SerializationFeature;
+import com.google.common.collect.ImmutableList;
+import com.google.common.util.concurrent.MoreExecutors;
+
+import com.linecorp.armeria.client.WebClient;
+import com.linecorp.armeria.common.AggregatedHttpResponse;
+import com.linecorp.armeria.common.HttpData;
+import com.linecorp.armeria.common.HttpObject;
+import com.linecorp.armeria.common.HttpResponse;
+import com.linecorp.armeria.common.HttpStatus;
+import com.linecorp.armeria.common.MediaType;
+import com.linecorp.armeria.common.ResponseHeaders;
+import com.linecorp.armeria.server.ServerBuilder;
+import com.linecorp.armeria.testing.junit5.server.ServerExtension;
+
+import io.netty.handler.codec.http.HttpHeaderNames;
+import reactor.core.publisher.Flux;
+import reactor.test.StepVerifier;
+
+public class JsonLinesTest {
+    private static final ObjectMapper mapper = new ObjectMapper();
+
+    @RegisterExtension
+    static ServerExtension server = new ServerExtension() {
+
+        @Override
+        protected void configure(ServerBuilder sb) {
+            sb.service("/seq/publisher",
+                    (ctx, req) -> JsonLines.fromPublisher(Flux.just("foo", "bar", "baz", "qux")))
+                    .service("/seq/stream",
+                            (ctx, req) -> JsonLines.fromStream(
+                                    Stream.of("foo", "bar", "baz", "qux"), MoreExecutors.directExecutor()))
+                    .service("/seq/custom-mapper",
+                            (ctx, req) -> JsonLines.fromPublisher(
+                                    Flux.just(new Pojo("Jon D", 21, Arrays.asList("Bmw", "Audi")),
+                                              new Pojo("Sarah D", 22, Arrays.asList("Tesla", "Honda"))),
+                                    new ObjectMapper().enable(SerializationFeature.INDENT_OUTPUT)))
+                    .service("/seq/single",
+                            (ctx, req) -> JsonLines.fromObject("foo"));
+            sb.disableServerHeader();
+            sb.disableDateHeader();
+        }
+    };
+
+    @Test
+    public void fromPublisherOrStreamMultiLineJson() {
+        final String record1 = "{\"name\":\"Jon D\",\"age\":21,\"cars\":[\"Bmw\",\"Audi\"]}\n";
+        final String record2 = "{\"name\":\"Sarah D\",\"age\":22,\"cars\":[\"Tesla\",\"Honda\"]}\n";
+        final WebClient client = WebClient.of(server.httpUri() + "/seq");
+        for (final String path : ImmutableList.of("/custom-mapper")) {
+            final HttpResponse response = client.get(path);
+            StepVerifier.create(response)
+                    .expectNext(ResponseHeaders.of(HttpStatus.OK,
+                            HttpHeaderNames.CONTENT_TYPE, MediaType.JSON_LINES))
+                    .assertNext(o ->
+                            ensureExpectedHttpData(o, record1, true))
+                    .assertNext(o ->
+                            ensureExpectedHttpData(o, record2, true))
```

Comment:
Since an `HttpData` could be split into multiple chunks, this could be a flaky test.
Could you aggregate the result and verify it?
Otherwise, `HttpDecoder` exists for it. For example:
https://github.com/line/armeria/blob/031bc8ee97d98709095a62077d3c3717c7be4fca/examples/server-sent-events/src/test/java/example/armeria/server/sse/MainTest.java#L92

Reply:
Made the changes.